framework/packages/utils/src/util.rs

```diff
@@ -28,7 +28,7 @@ pub fn coins_in_assets(assets: &[Asset]) -> Vec<Coin> {
             coins.push(Coin::new(asset.amount.u128(), denom.clone()));
         }
     }
-    // TODO: Hopefully some day tokenfactory coins can be last and we can get rid of this nonsense
-    coins.sort_unstable_by_key(|c| std::cmp::Reverse(c.denom.len()));
+    // https://github.com/cosmos/cosmos-sdk/blob/d5b40fc427f530b5ab078c61531d521b85505a1e/types/coin.go#L259-L261
+    coins.sort_unstable_by(|a, b| a.denom.cmp(&b.denom));
```

Comment:
At what level do we want that sorting to happen?

Reply:
Oh wow. Crazy that this is actually baked into the SDK as well. What do you mean which level?

Reply:
Agree, nevermind, it's gonna be too shady if we modify the message in any place.

Reply:
So we want this sort here? Why is it required?

Reply:
> So we want this sort here? Why is it required?

There is a link in a comment, so if coins not sorted transaction will fail